### DISCUSSION TITLE
Support of a workstation setup with a shared home directory

### DISCUSSION THREAD

**Discussion by dmentock** [spack/spack#28805] [General]
Our institution has a setup with multiple machines that can be individually accessed by our users. We would like to set up spack in a root directory such as `/opt` on each of those machines, and load it for each user by calling the `setup-env.sh` script after each login.

The user always uses the same home directory with the same `~/.spack` file, but accesses a different root directory where the spack directory is located, to install and load packages.
I tested this setup out by creating two copies of the spack repository on my machine (one in `/opt/test1` and one in `/opt/test2`). I called the `setup-env.sh` script in `test1` and installed and loaded a package and verified that it is in the `spack find --loaded` list. I then called the `setup-env.sh` script on `test2`, which is supposed to be "empty" with no packages installed yet, but found that `spack find --loaded` yields the same results and the package is already installed there as well. After logging out of the session and back in, can then select the other folder with it's own list of packages and load it accordingly.

For our setup, this would mean that a user installs a package on one machine, then swiches machines while maintaining the same home directory, and is then able to successfully load this package again on an entirely different spack folder.

As far as I understood, spack installs packages in the `spack/opt` directory. So in theory, if a user loaded the `setup-env.sh` of an empty spack directory he should not be able to access packages that are installed elsewhere. 
Does this pose a problem for our particular setup? How can a user be notified to reinstall a package that is currently loaded but not present on the machine he is using?

**Comment by haampie** [comment]
https://spack.readthedocs.io/en/latest/configuration.html#overriding-local-configuration

It wouldn't hurt to `export SPACK_USER_[CONFIG|CACHE]_PATH="$HOME/.spack/<machine name>"` instead of `~/.spack` to avoid subtle cache issues. It's fine to have the spack `opt` folder shared across machines

**Reply by dmentock** [reply]
Sorry for the confusion, the `spack/opt `folder is individual for each machine which is why I am making this post. The user can access different machines with different `spack/opt` directories from the same home directory with the same ~/.spack directory, so `spack/opt `changes while home stays the same. In our case, the problem would be that when a user chooses to switch machines, the package constellation suddenly changes and he needs to reload everything.

Is there any documentation on what information exactly is tracked in the ~/.spack directory? Installing and loading packages does not alter any of the information located there, where can I find where exactly the current installations and loaded modules tracked by spack?
